addons/webhooks/clusterbootstrap_webhook.go

```diff
@@ -292,7 +292,12 @@ func (wh *ClusterBootstrap) getGVR(gk schema.GroupKind) (*schema.GroupVersionRes
 	}
 	apiResourceList, err := wh.cachedDiscoveryClient.ServerPreferredResources()
 	if err != nil {
-		return nil, err
+		if apiResourceList == nil {
+			return nil, err
+		}
+		if _, ok := err.(*discovery.ErrGroupDiscoveryFailed); !ok {
+			return nil, err
+		}
```

Comment:
The comment by codecov is incorrect. The lines are covered by test added in this patch. Verified by stepping through with debugger